runtime/layers/fpm-dev/Dockerfile

```diff
@@ -1,11 +1,12 @@
 ARG PHP_VERSION
+ARG SKIP_EXTENSIONS=0
+
 FROM bref/build-php-$PHP_VERSION as build_extensions
 
-RUN pecl install xdebug
-RUN cp $(php -r "echo ini_get('extension_dir');")/xdebug.so /tmp
+COPY extensions.sh /tmp/extensions.sh
+RUN chmod +x /tmp/extensions.sh
+RUN /tmp/extensions.sh && rm /tmp/extensions.sh 
```

Comment:
Moving this into a separate script makes more sense than adding another stage IMO.

Using a separate bash script is also more maintainable and flexible in this scenario IMO.